### PR TITLE
OCPQE-26946: minimal permissions for both installer and ccoctl

### PR DESCRIPTION
**Pull request by jianli-wei**
- [OCPQE-26946 Add job for GCP sts(workload-identity) with minimal permissions](https://issues.redhat.com/browse/OCPQE-26946)
- Add workflow `cucushift-installer-rehearse-gcp-ipi-cco-manual-workload-identity-minimal-permission` for the testing scenario, which combines IPI with minimal permissions and IPI with CCO manual mode + workload-identity
- Update steps `ipi-conf-gcp-oidc-creds-provision` & `ipi-conf-gcp-oidc-creds-deprovision` to support env `ENABLE_MIN_PERMISSION_FOR_STS`
- Add one missing test, i.e. `gcp-ipi-oidc-mini-perm-rt-fips-amd-regen-cert-f28`, in 4.16 multi nightly
- Updated existing `gcp-ipi-oidc` tests (into `gcp-ipi-oidc-mini-perm` ones) for 4.15 and above

FYI What's updated:
YAML file keyword | Old Test | New Test
-- | -- | --
4.18__multi-nightly-4.18-upgrade-from-stable-4.17 | gcp-ipi-oidc-rt-fips-amd-f28 | gcp-ipi-oidc-mini-perm-rt-fips-amd-f28
4.18__multi-nightly-4.18-upgrade-from-stable-4.18 | gcp-ipi-oidc-rt-fips-amd-f60 | gcp-ipi-oidc-mini-perm-rt-fips-amd-f60
4.18__multi-nightly | gcp-ipi-oidc-rt-fips-amd-regen-cert-f14 | gcp-ipi-oidc-mini-perm-rt-fips-amd-regen-cert-f14
4.18__multi-nightly | gcp-ipi-oidc-arm-f28-destructive | gcp-ipi-oidc-mini-perm-arm-f28-destructive
4.17__multi-nightly-4.17-upgrade-from-stable-4.16 | gcp-ipi-oidc-rt-fips-amd-f28 | gcp-ipi-oidc-mini-perm-rt-fips-amd-f28
4.17__multi-nightly-4.17-upgrade-from-stable-4.17 | gcp-ipi-oidc-rt-fips-amd-f60 | gcp-ipi-oidc-mini-perm-rt-fips-amd-f60
4.17__multi-nightly | gcp-ipi-oidc-rt-fips-amd-regen-cert-f28 | gcp-ipi-oidc-mini-perm-rt-fips-amd-regen-cert-f28
4.17__multi-nightly | gcp-ipi-oidc-arm-f28-destructive | gcp-ipi-oidc-mini-perm-arm-f28-destructive
4.16__multi-nightly-4.16-upgrade-from-stable-4.15 | gcp-ipi-oidc-rt-fips-amd-f14 | gcp-ipi-oidc-mini-perm-rt-fips-amd-f14
4.16__multi-nightly-4.16-upgrade-from-stable-4.16 | gcp-ipi-oidc-rt-fips-amd-f14 | gcp-ipi-oidc-mini-perm-rt-fips-amd-f14
4.16__multi-nightly | gcp-ipi-oidc-arm-f28-destructive | gcp-ipi-oidc-mini-perm-arm-f28-destructive
4.15__multi-nightly-4.15-upgrade-from-stable-4.15 | gcp-ipi-oidc-rt-fips-amd-f28 | gcp-ipi-oidc-mini-perm-rt-fips-amd-f28
4.15__multi-nightly | gcp-ipi-oidc-rt-fips-amd-regen-cert-f28 | gcp-ipi-oidc-mini-perm-rt-fips-amd-regen-cert-f28
4.15__multi-nightly | gcp-ipi-oidc-arm-f28-destructive | gcp-ipi-oidc-mini-perm-arm-f28-destructive

### CODE REVIEW

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.17-installer-rehearse-gcp

**Comment by jianli-wei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-gcp-ipi-oidc-mini-perm-arm-f28-destructive

**Comment by jianli-wei** [comment]
@jianlinliu @gpei please help review, thanks!

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml [hunk 1]
```diff
@@ -1474,8 +1474,8 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: cucushift-installer-rehearse-gcp-ipi-disk-encryption
-- as: gcp-ipi-oidc-arm-f28-destructive
-  cron: 24 10 6 * *
+- as: gcp-ipi-oidc-mini-perm-arm-f28-destructive
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml:1477]: Sounds like we missed non-destructive job before, do you mind to add it together for now.

File: ci-operator/step-registry/ipi/conf/gcp/oidc-creds-provision/ipi-conf-gcp-oidc-creds-provision-commands.sh [hunk 2]
```diff
@@ -7,8 +7,15 @@ set -o pipefail
...
-export GCP_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/gce.json
-export GOOGLE_APPLICATION_CREDENTIALS="${GCP_SHARED_CREDENTIALS_FILE}"
+
+if [[ "${ENABLE_MIN_PERMISSION_FOR_STS}" == "true" ]]; then
+  echo "> Using minimal permissions for 'ccoctl'..."
+  export GCP_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/ccoctl_sa.json
+  export GOOGLE_APPLICATION_CREDENTIALS="${GCP_SHARED_CREDENTIALS_FILE}"
+else
+  export GCP_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/gce.json
+  export GOOGLE_APPLICATION_CREDENTIALS="${GCP_SHARED_CREDENTIALS_FILE}"
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/gcp/oidc-creds-provision/ipi-conf-gcp-oidc-creds-provision-commands.sh:17]: Seem like we can keep `GOOGLE_APPLICATION_CREDENTIALS` outside of the if block.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml [hunk 3]
```diff
@@ -1474,8 +1474,22 @@ tests:
     test:
     - chain: openshift-e2e-test-qe
     workflow: cucushift-installer-rehearse-gcp-ipi-disk-encryption
-- as: gcp-ipi-oidc-arm-f28-destructive
-  cron: 24 10 6 * *
+- as: gcp-ipi-oidc-mini-perm-rt-fips-amd-regen-cert-f28
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml:1477]:
```suggestion
- as: gcp-ipi-oidc-amd-f28
  cron: 39 13 17 * *
  steps:
    cluster_profile: gcp-qe
    env: 
      EXTRACT_MANIFEST_INCLUDED: "true"
    test:
    - chain: openshift-e2e-test-qe
    workflow: cucushift-installer-rehearse-gcp-ipi-cco-manual-workload-identity-minimal-permission
```

Comment (jianli-wei) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml:1477]: @jianlinliu I saw 4.15 / 4.17 / 4.18 are all with `gcp-ipi-oidc-mini-perm-rt-fips-amd-regen-cert-f28`, do you mean to let 4.16 alone with `gcp-ipi-oidc-mini-perm-fips-amd-f28`? 
```
$ grep -E "as: gcp-ipi-oidc" ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml
- as: gcp-ipi-oidc-mini-perm-rt-fips-amd-regen-cert-f28
- as: gcp-ipi-oidc-mini-perm-arm-f28-destructive
$ grep -E "as: gcp-ipi-oidc" ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__multi-nightly.yaml
- as: gcp-ipi-oidc-mini-perm-rt-fips-amd-regen-cert-f28
- as: gcp-ipi-oidc-mini-perm-arm-f28-destructive
$ grep -E "as: gcp-ipi-oidc" ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.18__multi-nightly.yaml
- as: gcp-ipi-oidc-mini-perm-rt-fips-amd-regen-cert-f14
- as: gcp-ipi-oidc-mini-perm-arm-f28-destructive
$ grep -E "as: gcp-ipi-oidc" ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml
- as: gcp-ipi-oidc-mini-perm-rt-fips-amd-regen-cert-f28
- as: gcp-ipi-oidc-mini-perm-arm-f28-destructive
$

Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__multi-nightly.yaml:1477]: okay. got it.

**Comment by jianlinliu** [comment]
cc @liangxia to review

**Pull request merged**